src/coord/info_command.c

```diff
@@ -162,21 +162,25 @@ static void convertField(InfoValue *dst, MRReply *src, InfoFieldType type) {
 }
 
 // Extract an array of FieldSpecInfo from MRReply
-void handleFieldStatistics(MRReply *src, InfoFields *fields) {
+void handleFieldStatistics(InfoFields *fields, MRReply *src, QueryError *error) {
   // Input validations
   RedisModule_Assert(src && fields);
   RedisModule_Assert(MRReply_Type(src) == MR_REPLY_ARRAY);
 
   size_t len = MRReply_Length(src);
   if (!fields->fieldSpecInfo_arr) {
     // Lazy initialization
-    fields->fieldSpecInfo_arr = array_new(FieldSpecInfo, len);
+    fields->fieldSpecInfo_arr = array_newlen(FieldSpecInfo, len);
     for (size_t i = 0; i < len; i++) {
-      FieldSpecInfo fieldSpecInfo = FieldSpecInfo_Init();
-      array_append(fields->fieldSpecInfo_arr, fieldSpecInfo);
+      fields->fieldSpecInfo_arr[i] = FieldSpecInfo_Init();
     }
   }
 
+  // Something went wrong (number of fields mismatch)
+  if (array_len(fields->fieldSpecInfo_arr) != len) {
+    return QueryError_SetCode(error, QUERY_EMISSMATCH);
```

Comment:
Consider
```suggestion
    QueryError_SetCode(error, QUERY_EMISSMATCH);
    return;
```
So it's clearer that nothing is returned.